mongoimport/mongoimport.go

```diff
@@ -556,11 +557,11 @@ func (up *upserter) Insert(doc interface{}) error {
 	var err error
 	if selector == nil { // modeInsert || doc-not-exist
 		log.Logvf(log.Info, "Could not construct selector from %v, falling back to insert mode", up.imp.upsertFields)
-		err = up.collection.Insert(document)
+		_, err = up.collection.InsertOne(nil, document, up.insertOpts)
 	} else if up.imp.IngestOptions.Mode == modeUpsert {
-		_, err = up.collection.Upsert(selector, document)
+		_, err = up.collection.ReplaceOne(nil, selector, document, up.replaceOpts)
 	} else { // modeMerge
-		_, err = up.collection.Upsert(selector, bson.M{"$set": document})
+		_, err = up.collection.UpdateOne(nil, selector, bson.M{"$set": document}, up.updateOpts)
```

Comment:
From my conversation with @xdg yesterday, this works because we've added MarshalBSON/UnmarshalBSON methods to mgo's bson.M and bson.D types so that the new driver can marshal/unmarshal them correctly.

Reply:
I like the idea of removing the dependency on mgo BSON for part of the mongoimport pipeline. One possible idea is to modify the readLoop in runInsertionWorker so that it takes each bson.D from the channel, converts that to a BSON document from the new driver, and pass the new document to Insert() so that the flushInserter interface works on types from the new driver rather than mgo. Thoughts?

Reply:
Ah, okay. Well, I would still say I'm in favor of limiting the usage of mgo's bson here to parsing the extJSON, assuming it wouldn't be a big change. Given that this does work as intended, if we're not dropping mgo bson altogether in this PR, then it's up to you if you think it's worth doing some of the work now as opposed to leaving it until later.

Reply:
I hadn't refreshed to see your second comment when I posted mine, so it was mainly in response to your previous one. But yeah that sounds good.

Reply:
Similar to my response on your other comment about the flushInserter interface: this seemed like a good idea but I now realize it would require changing BufferedBulkInserter in common as well.

Reply:
Dang. I guess we shall wait on TOOLS-2173 then. That ticket will be quite large.